analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/EmptyCollectionsShouldNotBeEnumeratedBase.cs

```diff
@@ -18,10 +18,131 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using System.Collections;
+using System.Collections.ObjectModel;
+using SonarAnalyzer.SymbolicExecution.Sonar.Constraints;
+
 namespace SonarAnalyzer.SymbolicExecution.Roslyn.RuleChecks;
 
 public abstract class EmptyCollectionsShouldNotBeEnumeratedBase : SymbolicRuleCheck
 {
     protected const string DiagnosticId = "S4158";
     protected const string MessageFormat = "Remove this call, the collection is known to be empty here.";
+
+    private static readonly ImmutableArray<KnownType> TrackedCollectionTypes = ImmutableArray.Create(
```

Comment:
Will you add nongeneric collections later too?

Reply:
Nongeneric collections are on my list of things to look into.

Reply:
And what the question about HashSet?